guides/keras_cv/simsiam-with-kerascv.py

```diff
@@ -0,0 +1,604 @@
+"""
+Title: SimSiam Training with TensorFlow Similarity and KerasCV
+Author: [lukewood](https://lukewood.xyz), Ian Stenbit, Owen Vallis
+Date created: 2022/09/21
+Last modified: 2022/09/21
+Description: Train a KerasCV model using unlabelled data with SimSiam.
+"""
+
+"""
+## Overview
+
+[TensorFlow similarity](https://github.com/tensorflow/similarity) makes it easy to train
+KerasCV models on unlabelled corpuses of data using contrastive learning algorithms such
+as SimCLR, SimSiam, and Barlow Twins.  In this guide, we will train a KerasCV model
+using the SimSiam implementation from TensorFlow Similarity.
+
+## Background
+
+Self-supervised learning is an approach to pre-training models using unlabeled data. T
+his approach drastically increases accuracy when you have very few labeled examples but
+a lot of unlabelled data.
+The key insight is that you can train a self-supervised model to learn data
+representations by contrasting multiple augmented views of the same example.
+These learned representations capture data invariants, e.g., object translation, color
+jitter, noise, etc. Training a simple linear classifier on top of the frozen
+representations is easier and requires fewer labels because the pre-trained model
+already produces meaningful and generally useful features.
+
+Overall, self-supervised pre-training learns representations which are more generic and
+robust than other approaches to augmented training and pre-training.  An overview of
+the general contrastive learning process is shown below:
+
+![Contrastive overview](https://i.imgur.com/mzaEq3C.png)
+
+In this tutorial, we will use the [SimSiam](https://arxiv.org/abs/2011.10566) algorithm
+for contrastive learning.  As of 2022, SimSiam is the state of the art algorithm for
+contrastive learning; allowing for unprecedented scores on CIFAR-100 and other datasets.
+
+To get started, we will sort out some imports.
```

Comment:
There are a couple of tutorials on the topic of SSL on keras.io. May be refer the readers to some of them?

Reply:
We can do this in the conclusion at the bottom